chore(batch): deprecate sqs_batch_processor

rubenfonseca opened the pull request:
<!-- markdownlint-disable MD041 MD043 -->
**Issue number:** #1459

## Summary

### Changes

> Please provide a summary of what's being changed

This PR adds a deprecation warning to the `sqs_batch_processor` decorator and `PartialSQSProcessor` class.

It also adds a banner on the documentation urging customers to upgrade before the next major version.

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [x] [Meet tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
* [x] I have performed a self-review of this change
* [ ] Changes have been tested
* [ ] Changes are documented
* [x] PR title follows [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-python/blob/develop/.github/semantic.yml)

<details>
<summary>Is this a breaking change?</summary>

**RFC issue number**:

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

rubenfonseca committed:
chore(batch): deprecate sqs_batch_processor

rubenfonseca commented:
@heitorlessa I do not know if the way I've implemented the banner on the docs is the correct way to do it.

heitorlessa commented on aws_lambda_powertools/utilities/batch/sqs.py:
```diff
@@ -77,6 +78,14 @@ def __init__(
 
+        warnings.warn(
+            "The sqs_batch_processor decorator and PartialSQSProcessor class are now deprecated, "
+            "and will be removed in the next major version of Powertools. "
```
```suggestion
            "and will be removed in the next major version. "
```

heitorlessa commented on docs/overrides/main.html:
```diff
@@ -1,5 +1,10 @@
 {% extends "base.html" %}
 
```
This would add an announcement for the entire documentation, we'd want under `batch.md` instead with admonition:

```
???+ warning
    our message...
```

This type of announcement **will** be super useful to let people know we're going to launch v2. So perhaps rephrase to inform them that? I think it'll also need to make it auto-hide so those who already seen don't see it again: https://squidfunk.github.io/mkdocs-material/setup/setting-up-the-header/?h=announcement#automatic-hiding

heitorlessa reviewed (changes requested):
Awesome! Two notes on repurposing announcement bar to inform customers of v2, and admonition for `batch.md`

rubenfonseca committed:
fix(docs): wording

Co-authored-by: Heitor Lessa <lessa@amazon.nl>

rubenfonseca committed:
docs(batch): change message placement

rubenfonseca commented:
Moved the warning to the batch doc page, and added a banner to documentation saying "v2 is coming". As always, feedback on wording is very important to me.

heitorlessa commented on docs/overrides/main.html:
```diff
@@ -1,5 +1,10 @@
 {% extends "base.html" %}
 
+{% block announce %}
+  ⚠️ Powertools for Python v2 is coming soon!
```
```suggestion
  👋 Powertools for Python v2 is coming soon!
```

heitorlessa replied:
@rubenfonseca that's the only thing I'd change. Wording is great!

👋 is a more friendly "attention grab" than ⚠️ that could be misunderstood as a problem

heitorlessa committed:
Update docs/overrides/main.html
